face/capture/src/main/java/com/simprints/face/capture/screens/FaceCaptureViewModel.kt

```diff
@@ -92,58 +95,64 @@ internal class FaceCaptureViewModel @Inject constructor(
+    ) = viewModelScope.launch {
         if (initialised) {
             Simber.i("Face bio SDK already initialised", tag = FACE_CAPTURE)
             return@launch
         }
+        this@FaceCaptureViewModel.bioSDK = sdk
 
         Simber.i("Starting face capture flow", tag = FACE_CAPTURE)
+        if (sdk == FaceConfiguration.BioSdk.RANK_ONE) {
+            val licenseVendor = Vendor.RankOne
+            val license = licenseRepository.getCachedLicense(licenseVendor)
+            var licenseStatus = license.determineLicenseStatus()
+            if (licenseStatus == LicenseStatus.VALID) {
+                licenseStatus = initialize(activity, license!!)
+            }
 
-        val licenseVendor = Vendor.RankOne
-        val license = licenseRepository.getCachedLicense(licenseVendor)
-        var licenseStatus = license.determineLicenseStatus()
-        if (licenseStatus == LicenseStatus.VALID) {
-            licenseStatus = initialize(activity, license!!)
-        }
-
-        // In some cases license is invalidated on initialisation attempt
-        if (licenseStatus != LicenseStatus.VALID) {
-            Simber.i("Face license is $licenseStatus - attempting download", tag = LICENSE)
-            licenseStatus = refreshLicenceAndRetry(
-                activity,
-                licenseVendor,
-                LicenseVersion(
-                    configManager
-                        .getProjectConfiguration()
-                        .face
-                        ?.rankOne
-                        ?.version
-                        .orEmpty(),
-                ),
-            )
-        }
-        // Still invalid after attempted refresh
-        if (licenseStatus != LicenseStatus.VALID) {
-            Simber.i("Face license is $licenseStatus", tag = LICENSE)
-            licenseRepository.deleteCachedLicense(Vendor.RankOne)
-            _invalidLicense.send()
+            // In some cases license is invalidated on initialisation attempt
+            if (licenseStatus != LicenseStatus.VALID) {
+                Simber.i("Face license is $licenseStatus - attempting download", tag = LICENSE)
+                licenseStatus = refreshLicenceAndRetry(
+                    activity,
+                    licenseVendor,
+                    LicenseVersion(
+                        configManager
+                            .getProjectConfiguration()
+                            .face
+                            ?.rankOne
+                            ?.version
+                            .orEmpty(),
+                    ),
+                )
+            }
+            // Still invalid after attempted refresh
+            if (licenseStatus != LicenseStatus.VALID) {
+                Simber.i("Face license is $licenseStatus", tag = LICENSE)
+                licenseRepository.deleteCachedLicense(Vendor.RankOne)
+                _invalidLicense.send()
+            }
+            saveLicenseCheckEvent(licenseVendor, licenseStatus)
```

Comment:
Maybe in the future we can abstract all this logic in the license repo, so that it takes the SDK as param and returns whatever license is appropriate and the call here is the same regardless of the SDK.

Reply:
I vote for removing the license module altogether. :D 

Made a ticket for it: https://simprints.atlassian.net/browse/MS-1011